Add some useful scripts

francescobrivio opened the pull request:
Adding few useful scripts to help creating and managing the PRs that update the autoCond.py script.
The scripts were originally written by @christopheralanwest and slightly updated to take into account the latest changes in the GTs and queues.

**checkoutIB.sh**
Downloads the latest available IB in a release cycle, invoked by:
`scripts/checkoutIB.sh [RELEASE CYCLE]`
for example: _scripts/checkoutIB.sh 11_3_X_

**diff_queues.py**
Performs the difference between the global tags listed in autoCond and the corresponding queues and prints to screen the differences. Invoked by:
`python scripts/diff_queues.py`

**diff_changes.py**
Performs the difference between the GTs in your CMSSW release directory (including the modifications) with those in $CMSSW_RELEASE_BASE and prints out a list of URLs showing the set of GT diffs for use in a PR description.
Invoked by:
`python scripts/diff_changes.py`

mmusich commented:
@francescobrivio I invited you to join the repo as owner, so you can merge your own PRs.

mmusich commented:
It might be useful to add an help function.
I tried for curiosity one:
```bash
$ python diff_changes.py 
diff: /tmp/musich/CMSSW_11_3_X_2021-02-28-2300/src/Configuration/AlCa/python/autoCond.py: No such file or directory
diff: /tmp/musich/CMSSW_11_3_X_2021-02-28-2300/src/Configuration/AlCa/python/autoCond.py: No such file or directory
Traceback (most recent call last):
  File "diff_changes.py", line 112, in <module>
    main()
  File "diff_changes.py", line 86, in main
    if int(gt_base.split('_')[0][0:2]) < 111:
ValueError: invalid literal for int() with base 10: ''
```
so it presumes you checked out the package.
After adding the package:
```bash
$ git cms-addpkg Configuration/AlCa/
$ python diff_changes.py
Traceback (most recent call last):
  File "diff_changes.py", line 112, in <module>
    main()
  File "diff_changes.py", line 86, in main
    if int(gt_base.split('_')[0][0:2]) < 111:
ValueError: invalid literal for int() with base 10: ''
```

francescobrivio commented:
> It might be useful to add an help function.
> I tried for curiosity one:
> 
> ```shell
> $ python diff_changes.py 
> diff: /tmp/musich/CMSSW_11_3_X_2021-02-28-2300/src/Configuration/AlCa/python/autoCond.py: No such file or directory
> diff: /tmp/musich/CMSSW_11_3_X_2021-02-28-2300/src/Configuration/AlCa/python/autoCond.py: No such file or directory
> Traceback (most recent call last):
>   File "diff_changes.py", line 112, in <module>
>     main()
>   File "diff_changes.py", line 86, in main
>     if int(gt_base.split('_')[0][0:2]) < 111:
> ValueError: invalid literal for int() with base 10: ''
> ```
> 
> so it presumes you checked out the package.
> After adding the package:
> 
> ```shell
> $ git cms-addpkg Configuration/AlCa/
> $ python diff_changes.py
> Traceback (most recent call last):
>   File "diff_changes.py", line 112, in <module>
>     main()
>   File "diff_changes.py", line 86, in main
>     if int(gt_base.split('_')[0][0:2]) < 111:
> ValueError: invalid literal for int() with base 10: ''
> ```

Indeed I will try to add some checks to avoid these errors.
I also noticed that the description of the `diff_changes.py` script is a copy paste from the other script so I will fix that.

francescobrivio commented:
> It might be useful to add an help function.
> I tried for curiosity one:
> 
> ```shell
> $ python diff_changes.py 
> diff: /tmp/musich/CMSSW_11_3_X_2021-02-28-2300/src/Configuration/AlCa/python/autoCond.py: No such file or directory
> diff: /tmp/musich/CMSSW_11_3_X_2021-02-28-2300/src/Configuration/AlCa/python/autoCond.py: No such file or directory
> Traceback (most recent call last):
>   File "diff_changes.py", line 112, in <module>
>     main()
>   File "diff_changes.py", line 86, in main
>     if int(gt_base.split('_')[0][0:2]) < 111:
> ValueError: invalid literal for int() with base 10: ''
> ```
> 
> so it presumes you checked out the package.
> After adding the package:
> 
> ```shell
> $ git cms-addpkg Configuration/AlCa/
> $ python diff_changes.py
> Traceback (most recent call last):
>   File "diff_changes.py", line 112, in <module>
>     main()
>   File "diff_changes.py", line 86, in main
>     if int(gt_base.split('_')[0][0:2]) < 111:
> ValueError: invalid literal for int() with base 10: ''
> ```

I added a function (`checkEnvironment`) that automatically checks if the scripts are run under a CMSSW release and if the `Configuration/AlCa` package has been checked out.
I'm not sure I understand what is the second error that you got: the error message is exactly the same as before you checked out the package.

mmusich commented:
>  I'm not sure I understand what is the second error that you got: the error message is exactly the same as before you checked out the package.

indeed I am still getting it with the new version...

mmusich commented on scripts/diff_changes.py:
```diff
@@ -0,0 +1,132 @@
+
+    for new_gt in new_gts:
+        gt_base = re.sub('_v[0-9]*', '_', new_gt)
+        if int(gt_base.split('_')[0][0:2]) < 111:
```
so if one doesn't change anything in the local installation, the script will end in error.

francescobrivio replied:
Indeed, I just managed to reproduce the error. I will fix it. Thanks for spotting it!

francescobrivio commented:
@mmusich not sure if you had time to test this but I think now all needed protections are in place. I will merge it.

mmusich commented:
@francescobrivio I didn't have time yet, but then again I didn't know I was supposed to.
If you wish me to review you can assign the PR to me (by attaching an assignee) and I'll do my best to check it.

francescobrivio commented:
> @francescobrivio I didn't have time yet, but then again I didn't know I was supposed to.
> If you wish me to review you can assign the PR to me (by attaching an assignee) and I'll do my best to check it.

No it's fine! It was more an heads up that I was going to merge it. Thanks again for the suggestions!
